File: CS214/projects/proj03/LetterGrade.java
/* LetterGrade.java is a driver for function letterGrade().
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Ryan Vreeke
 * Date: 25 Feb, 2020
 ****************************************************************/

import java.util.Scanner;

public class LetterGrade {

   public static void main(String[] args) {
      Scanner keyboard = new Scanner(System.in);
      System.out.println("\nEnter your average: "); // prompt

      int average = keyboard.nextInt();      // read average
      System.out.println(letterGrade(average)); // display its code
   }

   /***************************************************************
    * letterGrade() converts average into its corresponding letter grade.
    *
    * Receive: average, an integer.
    * PRE: average is positive
    * Return: the letter grade corresponding to average.
    */

    public static char letterGrade(int average) {
        switch (average / 10) {
            case 10: case 9:
                return 'A';
            case 8:
                return 'B';
            case 7: 
                return 'C';
            case 6:
                return 'D';
            default:
                return 'F';
        }
   }

}

File: CS214/projects/proj03/clojure/src/letter_grade.clj
;;;; letter_grade.clj is a driver for function letterGrade().
;;;;
;;;; Input: average an integer
;;;; Output: The character corresponding to the average
;;;;          could be 'A', 'B', 'C', 'D', or 'F'
;;;;
;;;; Usage: clojure -m letter_grade
;;;;
;;;; Begun by: Prof. Adams, for CS 214 at Calvin College.
;;;; Completed by: Ryan Vreeke
;;;; Date: 25 Feb, 2020
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns letter_grade)                    ; name the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; letterGrade() returns the code for a given academic year.
;;; Receive: average, an integer.
;;; Precondition: average is positive
;;; Return: the character corresponding to the average.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn letterGrade [average] 
  (cond 
    (= (quot average 10) 10) \A
    (= (quot average 10) 9)  \A
    (= (quot average 10) 8)  \B
    (= (quot average 10) 7)  \C
    (= (quot average 10) 6)  \D
    :else                    \F
  )
)

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function -main() test-drives our letterGrade() function.
;;; Input: average: an integer.
;;; Output: the corresponding letter grade 'A', 'B', 'C', 'D', or 'F'.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
 (print "\nEnter your average: ") (flush)
 (let
    [ average (read) ]                   ; read average (an integer)
    (println (letterGrade average))      ; display its code
 )
)

